File: drills/elixir/harnesses/nc149_max_path_sum.exs
# Trees are written out directly: nil for empty, {value, left, right} for a
# node. `bud` is a node with no children.
bud = fn value -> {value, nil, nil} end

[
  {"max_path_sum(1/2/3)", inspect(6),
   inspect(Solution.max_path_sum({1, bud.(2), bud.(3)}))},
  {"max_path_sum(-10/9/20)", inspect(42),
   inspect(Solution.max_path_sum({-10, bud.(9), {20, bud.(15), bud.(7)}}))},
  {"max_path_sum(a single negative node)", inspect(-3),
   inspect(Solution.max_path_sum(bud.(-3)))},
  {"max_path_sum(all negative)", inspect(-1),
   inspect(Solution.max_path_sum({-2, bud.(-1), nil}))},
  {"max_path_sum(a single zero)", inspect(0),
   inspect(Solution.max_path_sum(bud.(0)))}
]

File: drills/go/harnesses/nc149_max_path_sum.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("maxPathSum([1,2,3])", 6, maxPathSum(tree(1, 2, 3))),
			tc("maxPathSum([-10,9,20,null,null,15,7])", 42, maxPathSum(tree(-10, 9, 20, x, x, 15, 7))),
			tc("maxPathSum([-3]) -- a single negative node", -3, maxPathSum(tree(-3))),
			tc("maxPathSum([-2,-1]) -- all negative", -1, maxPathSum(tree(-2, -1))),
			tc("maxPathSum([0])", 0, maxPathSum(tree(0))),
		}
	})
}
